### PR TITLE
Update numpy to version 1.20.2

### PR DESCRIPTION
**Pull request by misl6**
- Upgraded numpy from **1.16.4** to **1.20.2**
- Added a `hostpython_prerequisites` recipe configuration. We're now able to install hostpython prerequisites! (In this case, Cython was needed)

Could potentially fix:
- Issue #603 (build was failing due to an incompatibility between python3.9.2 and numpy 1.16.4)
- Issue reported by user "tdmartin4" on Discord.

I did some basic testing, without any test release to the App Store. In past (see issue #409) , some **numpy**  were named at the same way as **Apple Non-public API functions**. (For example `xerbla`). Seems that this function now follows a new naming logic, so I removed the related patch that is outdated and potentially un-necessary.

Feel free to open a separate issue if `xerbla` or other functions are now offending the App Store validation.

### CODE REVIEW

**Comment by misl6** [comment]
Ok, seems that the CI (that is based on MacOS Catalina) is failing.

The fix shouldn't be too hard. Will look at that.

BigSur is fine.

File: kivy_ios/toolchain.py [hunk 1]
```diff
@@ -757,6 +759,11 @@ def extract_arch(self, arch):
             ensure_dir(build_dir)
             self.extract_file(self.archive_fn, build_dir)
 
+    @cache_execution
+    def install_hostpython_prerequisites(self):
+        for prerequisite in self.hostpython_prerequisites:
+            _hostpython_pip(["install", prerequisite])
```
Comment (AndreMiras) [inline kivy_ios/toolchain.py:765]: How about passing the prerequisites list directly?
```suggestion
        _hostpython_pip(["install", self.hostpython_prerequisites])
```
You get the idea

File: kivy_ios/toolchain.py [hunk 2]
```diff
@@ -1171,6 +1178,14 @@ def _pip(args):
     shprint(pip_cmd, *args, _env=pip_env)
 
 
+def _hostpython_pip(args):
+    ctx = Context()
+    pip_path = join(ctx.dist_dir, 'hostpython3', 'bin', 'pip3')
+    logger.error("Executing pip for hostpython with: {}".format(args))
```
Comment (AndreMiras) [inline kivy_ios/toolchain.py:1184]: Why `logger.error` rather than `logger.info`?

**Review by AndreMiras — COMMENTED** [review]
LGTM, feel free to ping me when you're happy with the CI fixes

File: kivy_ios/toolchain.py (hunk 2 above)
Comment (misl6) [reply kivy_ios/toolchain.py:1184]: I assume I copy-pasted from here (I already did a git blame and guess Who's the author 🤦  ):
https://github.com/kivy/kivy-ios/blob/6b245b102dbac76428f9807f25056c9e1bd53ce6/kivy_ios/toolchain.py#L1169

Fixed both. Will push within the CI fixes.

File: kivy_ios/toolchain.py (hunk 1 above)
Comment (misl6) [reply kivy_ios/toolchain.py:765]: The suggestion looked great at a first glance but after getting it a try,  if `self.hostpython_prerequisites == []`(that is the default), the install function crashes badly.

I could opt for a set of conditions, but I feel comfortable with this `for` loop, but let me know what do you think about it.

Comment (AndreMiras) [reply kivy_ios/toolchain.py:765]: Yeah it was a quick one. We could obviously keep the for loop or go with some fancy things.
```python
self.hostpython_prerequisites and _hostpython_pip(["install", self.hostpython_prerequisites])
```
Or a proper if
```
if self.hostpython_prerequisites:
    _hostpython_pip(["install", self.hostpython_prerequisites])
```
Or even simply check for `args` in `_hostpython_pip()` and return if empty.
But yeah keeping the for loop is completely fine

File: kivy_ios/toolchain.py (hunk 2 above)
Comment (AndreMiras) [reply kivy_ios/toolchain.py]: Ahah don't blame the contributor, blame the reviewer :sweat_smile: 
Thanks for the fix and yeah I also love git blaming :smile:

**Review by AndreMiras — APPROVED** [review]
Sweet :heart:

**Pull request merged**